Content.Server/Radio/EntitySystems/HeadsetSystem.cs

```diff
@@ -103,4 +104,16 @@ private void OnExamined(EntityUid uid, HeadsetComponent component, ExaminedEvent
 
         args.PushMarkup(Loc.GetString("examine-headset-chat-prefix", ("prefix", ";")));
     }
+
+    public void JamChannel(string channel) {
+        foreach (var transmitter in EntityQuery<HeadsetComponent>(true)) {
+            transmitter.JammedChannels.Add(channel);
+        }
+    }
+
+    public void UnJamChannel(string channel) {
+        foreach (var transmitter in EntityQuery<HeadsetComponent>(true)) {
+            transmitter.JammedChannels.Remove(channel);
+        }
+    }
```

Comment:
It'd be much, much better if you generalized this jamming system to fire off a "check if channel jammed" broadcast event that includes the transmitting entity.

What you have here breaks if they simply get a new headset from a vending machine.

Reply:
> It'd be much, much better if you generalized this jamming system to fire off a "check if channel jammed" broadcast event that includes the transmitting entity.
> 
> What you have here breaks if they simply get a new headset from a vending machine.

Like make an event in /Radio and fire it before sending/receiving a message + subscribe SolarFlare to the event (as it's just an EntitySystem)? I will try this a bit later.

Reply:
Actually I can just use RadioSendAttemptEvent and cancel it.